geoprocessing/src/modules/shapefiles/shapefiles.service.ts

```diff
@@ -0,0 +1,56 @@
+import { Injectable, Logger } from '@nestjs/common';
+
+import { createReadStream, rmdirSync } from 'fs';
+const mapshaper = require('mapshaper');
+const unzipper = require('unzipper');
+
+@Injectable()
+export class ShapeFileService {
+  private readonly logger: Logger = new Logger(ShapeFileService.name);
+  constructor() {}
+
+  private _unzipShapefile(fileInfo: Express.Multer.File): Promise<string> {
+    return new Promise((resolve, reject) => {
+      createReadStream(fileInfo.path)
+        .pipe(
+          unzipper.Extract({
+            path: fileInfo.destination + fileInfo.filename.replace('.zip', ''),
+          }),
+        )
+        .on('close', () =>
+          resolve(`${fileInfo.filename} extracted succesfully`),
+        )
+        .on('error', () =>
+          reject(new Error(`${fileInfo.filename} could not been extracted`)),
+        );
+    });
+  }
+
+  private async _shapeFileToGeoJson(fileInfo: Express.Multer.File) {
+    const outputKey = `shapefile-${new Date().getTime()}.gejson`;
+    const _geoJson = await mapshaper.applyCommands(
+      `-i ${fileInfo.path.replace('.zip', '')}/*.shp -o ${outputKey}`,
+    );
+
+    return JSON.parse(_geoJson[outputKey].toString('utf-8'));
+  }
+
+  private wipeOutFolder(path: string): void {
+    rmdirSync(path, { recursive: true });
```

Comment:
I am always a bit nervous about `rm -rf` without checks on the path - I would either throw an error if path doesn't start with `/tmp` plus something else, or accept the *name* (not the path) of a temp folder, and prepend `/tmp/whatever` here.

Reply:
@hotzevzl I see the point. In this case multer will add the destination folder according to what we decided to pass in `uploadOptions`
I'll add yout first suggestion